test/e2e_node/dockershim_checkpoint_test.go

```diff
@@ -0,0 +1,210 @@
+package e2e_node
+
+import (
+	"crypto/md5"
+	"fmt"
+	"os"
+	"os/exec"
+	"regexp"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/pkg/api/v1"
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+const (
+	testPod                 = "checkpoint-test-pod"
+	testCheckpoint          = "checkpoint-test"
+	dockershimCheckpointDir = "/var/lib/dockershim/sandbox"
+	gcTimeout               = 3 * time.Minute
+	testCheckpointContent   = `{"version":"v1","name":"fluentd-gcp-v2.0-vmnqx","namespace":"kube-system","data":{},"checksum":1799154314}`
+)
+
+var _ = framework.KubeDescribe("Dockershim [Serial] [Disruptive]", func() {
+	f := framework.NewDefaultFramework("dockerhism-checkpoint-test")
+
+	It("should clean up pod sandbox checkpoint after pod deletion", func() {
```

Comment:
I don't think this test check whether the checkpoint was created in the first place. Either add the check or create a dedicated test for it.